### ISSUE TITLE
Fixing portability issues with Squeak 5.1 on Seaside 3.1

### ISSUE THREAD

**Issue by gettimothy** [astares/Seaside-Bootstrap#2]
Symptoms: Clicking the "Browse" button on the examples throws a DNU. Proceeding allows the examples to show up, however, there is no source code in the "source code" box.

Tobias Pape suggested this fix as documented here: http://forum.world.st/unregistered-not-called-on-expired-session-td5070103.html#a5070624 

TL;DR

Add class side method to

```
TBSBootstrapExample class >>methodNamed: aSymbol
^ self >> aSymbol

```
Add instance side method to:

```
CompiledMethod >sourceCode
^ self getSource
```

**Comment by astares** [comment]
Thanks. I never tested this one on Squeak. 

I added the #methodNamed: in Bootstrap-Examples-TorstenBergmann.58 for compatibility.

While I was able to add #methodNamed: I will not add the sourceCode method calling getSource
as #getSource is deprecated in Pharo and should not be called.

Fixing this in the TBS project would not make sense as this is a compatibility problem between
Squeak and Pharo. Pharo already uses "sourceCode" as we want to avoid Java like "getXXX" methods and Squeak has not yet catched up.

I would suggest to deprecate #getSource in Squeak too and rather have a #sourceCode accessor there too (either with an own extension in your app code for older Squeak or directly for the base system
in recent Squeak)

**Comment by astares** [comment]
To be found in maintenance release 0.17.1

**Comment by krono** [comment]
Hi @astares, maybe there's already something in Grease that can be used. Seaside itself sometimes needs those messages, so we're maybe just lucky…

**Comment by gettimothy** [comment]
Thank you. I will check Grease oer Tobias. Failing that, the squeak-dev folks might want to follow uo




---- On Thu, 15 Mar 2018 03:47:38 -0400 notifications@github.com wrote ---- Thanks. I never tested this one on Squeak. I added the #methodNamed: in Bootstrap-Examples-TorstenBergmann.58 for compatibility. While I was able to add #methodNamed: I will not add the sourceCode method calling getSource
 as #getSource is deprecated in Pharo and should not be called. Fixing this in the TBS project would not make sense as this is a compatibility problem between
 Squeak and Pharo. Pharo already uses "sourceCode" as we want to avoid Java like "getXXX" methods and Squeak has not yet catched up. I would suggest to deprecate #getSource in Squeak too and rather have a #sourceCode accessor there too (either with an own extension in your app code for older Squeak or directly for the base system
 in recent Squeak)  —
